forms-flow-bpm/forms-flow-bpm-camunda/src/main/java/org/camunda/bpm/extension/commons/utils/VaultConfig.java

```diff
@@ -0,0 +1,57 @@
+package org.camunda.bpm.extension.commons.utils;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.vault.core.VaultTemplate;
+import org.springframework.vault.support.VaultResponse;
+import org.springframework.vault.core.VaultKeyValueOperationsSupport.KeyValueBackend;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Configuration
+public class VaultConfig {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(VaultConfig.class);
+
+    private static VaultTemplate vaultTemplate;
+    private static String vaultPath;
+    private static String vaultSecret;
+
+    @Autowired(required = false)
+    public void setVaultTemplate(VaultTemplate template) {
+        vaultTemplate = template;
+    }
+
+    @Value("${spring.cloud.vault.path}")
+    public void setVaultPath(String path) {
+        vaultPath = path;
+    }
+
+    @Value("${spring.cloud.vault.secret}")
+    public void setVaultSecret(String secret) {
+        vaultSecret = secret;
+    }
+
+    public static String getSecret() {

```

Comment:
Can this be called statically for any listener ?

Reply:
yes. Tried in the BPMFormDataPipelineListener
![image](https://github.com/AOT-Technologies/forms-flow-ai/assets/99173163/3f498a9c-7275-4f0e-8da7-7b1f8ee055bb)